extensions/binding/org.eclipse.smarthome.binding.bluetooth/src/main/java/org/eclipse/smarthome/binding/bluetooth/BluetoothCharacteristic.java

```diff
@@ -0,0 +1,633 @@
+     */
+    public int getWriteType() {
+        return writeType;
+    }
+
+    /**
+     * Set the write type for this characteristic
+     *
+     * @param writeType
+     */
+    public void setWriteType(int writeType) {
+        this.writeType = writeType;
+    }
+
+    /**
+     * Get the service to which this characteristic belongs
+     *
+     * @return the {@link BluetoothService}
+     */
+    public BluetoothService getService() {
+        return service;
+    }
+
+    /**
+     * Returns the handle for this characteristic
+     *
+     * @return the handle for the characteristic
+     */
+    public int getHandle() {
+        return handle;
+    }
+
+    /**
+     * Get the service to which this characteristic belongs
+     *
+     * @return the {@link BluetoothService}
+     */
+    public void setService(BluetoothService service) {
+        this.service = service;
+    }
+
+    /**
+     * Returns a list of descriptors for this characteristic.
+     *
+     */
+    public List<BluetoothDescriptor> getDescriptors() {
+        return new ArrayList<BluetoothDescriptor>(gattDescriptors.values());
+    }
+
+    /**
+     * Returns a descriptor with a given UUID out of the list of
+     * descriptors for this characteristic.
+     *
+     * @return the {@link BluetoothDescriptor}
+     */
+    public BluetoothDescriptor getDescriptor(UUID uuid) {
+        return gattDescriptors.get(uuid);
+    }
+
+    /**
+     * Get the stored value for this characteristic.
+     *
+     */
+    public int[] getValue() {
+        return value;
+    }
+
+    /**
+     * Get the stored value for this characteristic.
+     *
+     */
+    public byte[] getByteValue() {
+        byte[] byteValue = new byte[value.length];
+        for (int cnt = 0; cnt < value.length; cnt++) {
+            byteValue[cnt] = (byte) (value[cnt] & 0xFF);
+        }
+        return byteValue;
+    }
+
+    /**
+     * Return the stored value of this characteristic.
+     *
+     */
+    public Integer getIntValue(int formatType, int offset) {
+        if ((offset + getTypeLen(formatType)) > value.length) {
+            return null;
+        }
+
+        switch (formatType) {
+            case FORMAT_UINT8:
+                return unsignedByteToInt(value[offset]);
+
+            case FORMAT_UINT16:
+                return unsignedBytesToInt(value[offset], value[offset + 1]);
+
+            case FORMAT_UINT32:
+                return unsignedBytesToInt(value[offset], value[offset + 1], value[offset + 2], value[offset + 3]);
+
+            case FORMAT_SINT8:
+                return unsignedToSigned(unsignedByteToInt(value[offset]), 8);
+
+            case FORMAT_SINT16:
+                return unsignedToSigned(unsignedBytesToInt(value[offset], value[offset + 1]), 16);
+
+            case FORMAT_SINT32:
+                return unsignedToSigned(
+                        unsignedBytesToInt(value[offset], value[offset + 1], value[offset + 2], value[offset + 3]), 32);
+        }
```

Comment:
woudln't it make sense to have a `default` case as well, to let it fail loudly in case somebody extends it?